site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx

```diff
@@ -144,7 +172,10 @@ export const WorkspaceBuildPageView: FC<WorkspaceBuildPageViewProps> = ({
 					))}
 				</Sidebar>
 
-				<div css={{ height: "100%", overflowY: "auto", width: "100%" }}>
+				<div
+					ref={contentRef}
+					css={{ height: "100%", overflowY: "auto", width: "100%" }}
```

Comment:
I think I'd do something like this:
```tsx
const [contentHeight, setContentHeight] = useState<string>()
const contentRef = useRef<HTMLDivElement>(null);
useLayoutEffect(() => {
  // It's been a while since I've had to do this, but dispatching a state update from the
  // resize callback might actually be too slow, and you might get a UI flicker still. You 
  // might need to dispatch the state update AND mutate the DOM node directly when the
  // effect first mounts
}, [])

css={{ height: contentHeight || "100%" }}
```